[system] Add `experimental_sx` utility (#29833)
mui-org/material-ui: docs/src/pages/system/styled/styled.md

})
```

> ### How can I use the `sx` syntax with the `styled()` utility?

Reviewer: Should we note somewhere that there will be an overhead compared to not using `sx`?

Author: The overhead is the same as if you were to use the `sx` prop, no? If we state this, we should have some numbers to be more concrete.

Reviewer: > The overhead is the same as if you were to use the sx prop

Yeah, we can add the numbers later then.

Author: I've added a note that it is the same as if the `sx` prop was used. I guess it should be enough.

})
```

### How can I use the `sx` syntax with the `styled()` utility?

> If you are one of those that prefer the `sx` syntax and want to use it in both the `sx` prop and the `styled()` utility, you can do this by using the `experimental_sx` utility from the `@mui/system`:

Reviewer 2: ```suggestion
If you are one of those who prefers the `sx` syntax and wants to use it in both the `sx` prop and the `styled()` utility, you can use the `experimental_sx` utility from the `@mui/system`:
```

{{"demo": "pages/system/styled/UsingWithSx.js", "defaultCodeOpen": true}}

The overhead added by using the `experimental_sx` utility is the same as if you were to use the `sx` prop on the component.

> > Note: You can use the `experimental_sx` outside of the `styled()` utility too, for example for defining the `variants` in your custom theme.

Reviewer 2: ```suggestion
> Note: You can use the `experimental_sx` outside of the `styled()` utility, too; e.g., for defining the `variants` in your custom theme.
```
